src/AzureAuth.Test/CommandMainTest.cs

```diff
@@ -644,6 +644,63 @@ public void TestGenerateEvent_From_AuthFlowResult_With_TokenResult_And_Null_Erro
             eventData.Measures.Should().ContainKey("duration_milliseconds");
         }
 
+        /// <summary>
+        /// Test to validate user interactive env.
+        /// </summary>
+        [Test]
+        public void InteractivityIsNotDisabledIfCorextIsSetToOne()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            this.envMock.Setup(e => e.Get("Corext_NonInteractive")).Returns("1");
+            subject.InteractivityDisabled().Should().BeTrue();
+        }
+
+        /// <summary>
+        /// Test to validate user interactive env.
+        /// </summary>
+        [Test]
+        public void InteractivityIsNotDisabledIfCorextIsNotSetToOneOrTrue()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            this.envMock.Setup(e => e.Get("Corext_NonInteractive")).Returns("random-value");
+            subject.InteractivityDisabled().Should().BeFalse();
+        }
+
+        /// <summary>
+        /// Test to validate user interactive env.
+        /// </summary>
+        [Test]
+        public void InteractivityIsDisabledIfEnvVarIsSetToNonEmpty()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            this.envMock.Setup(e => e.Get("AZUREAUTH_DISABLE_INTERACTIVITY")).Returns("1");
+            subject.InteractivityDisabled().Should().BeTrue();
+
+            this.envMock.Setup(e => e.Get("AZUREAUTH_DISABLE_INTERACTIVITY")).Returns("non-empty-string");
+            subject.InteractivityDisabled().Should().BeTrue();

```

Comment:
Within a given test - it's best practice not to re-use the subject more than once, in order to test the behavior under isolation. This could be done with a `TestCase` based test.